esx_service/vmdk_ops.py

```diff
@@ -749,9 +749,11 @@ def executeRequest(vm_uuid, vm_name, config_path, cmd, full_vol_name, opts):
                                   tenant_uuid=tenant_uuid,
                                   datastore=datastore)
         elif cmd == "attach":
-            response = attachVMDK(vmdk_path, vm_uuid)
+            with lockManager.get_lock(vm_uuid):
+                response = attachVMDK(vmdk_path, vm_uuid)
         elif cmd == "detach":
-            response = detachVMDK(vmdk_path, vm_uuid)
+            with lockManager.get_lock(vm_uuid):
+                response = detachVMDK(vmdk_path, vm_uuid)
```

Comment:
Disk level locks are good. :)
The error comes from already having a reconfig task running for the same VM. (different volumes)
I didn't investigated why the mutex in the plugin driver is not protecting this situation, but serialization for attach/detach is really needed on the esx part. I might be missing something, but in the plugin we need to protect the refcount part only?